src/main/java/org/jabref/logic/exporter/EndnoteXmlExporter.java

```diff
@@ -0,0 +1,209 @@
+        writeField(xml, "number", entry.getField(StandardField.NUMBER).orElse(""), null);
+        writeField(xml, "dates", null, Map.of(), entry.getField(StandardField.YEAR).orElse(""), "year");
+        writePublisher(entry, xml);
+        writeField(xml, "isbn", entry.getField(StandardField.ISBN).orElse(""), null);
+        writeField(xml, "abstract", entry.getField(StandardField.ABSTRACT).orElse(""), null);
+        writeField(xml, "notes", entry.getField(StandardField.NOTE).orElse(""), null);
+        writeField(xml, "urls", null, Map.of(), entry.getField(StandardField.URL).orElse(""), "web-urls");
+        writeField(xml, "electronic-resource-num", entry.getField(StandardField.DOI).orElse(""), null);
+
+        xml.writeEndElement(); // record
+    }
+
+    private void writeField(XMLStreamWriter xml, String name, String value, Map<String, String> attributes) throws XMLStreamException {
+        if (value != null && !value.isEmpty()) {
+            xml.writeStartElement(name);
+            if (attributes != null) {
+                for (Map.Entry<String, String> attribute : attributes.entrySet()) {
+                    xml.writeAttribute(attribute.getKey(), attribute.getValue());
+                }
+            }
+            xml.writeStartElement("style");
+            xml.writeAttribute("face", "normal");
+            xml.writeAttribute("font", "default");
+            xml.writeAttribute("size", "100%");
+            xml.writeCharacters(value);
+            xml.writeEndElement(); // style
+            xml.writeEndElement(); // name
+        }
+    }
+
+    private void writeField(XMLStreamWriter xml, String name, String value, Map<String, String> attributes, String childValue, String childElementName) throws XMLStreamException {
+        xml.writeStartElement(name);
+        if (attributes != null) {
+            for (Map.Entry<String, String> attribute : attributes.entrySet()) {
+                xml.writeAttribute(attribute.getKey(), attribute.getValue());
+            }
+        }
+        if (childValue != null && !childValue.isEmpty()) {
+            xml.writeStartElement(childElementName);
+            xml.writeStartElement("style");
+            xml.writeAttribute("face", "normal");
+            xml.writeAttribute("font", "default");
+            xml.writeAttribute("size", "100%");
+            xml.writeCharacters(childValue);
+            xml.writeEndElement(); // style
+            xml.writeEndElement(); // childElementName
+        }
+        xml.writeEndElement(); // name
+    }
+
+    private void writeContributors(BibEntry entry, XMLStreamWriter xml) throws XMLStreamException {
+        entry.getField(StandardField.AUTHOR).ifPresent(authors -> {
+            try {
+                xml.writeStartElement("contributors");
+                xml.writeStartElement("authors");
+                for (Author author : authorListParser.parse(authors).getAuthors()) {
```

Comment:
nitpick: I feel this line is doing a lot of things, can you simplify it a bit?

Reply:
I made a tiny separation, see if this makes it better.